<a href="https://colab.research.google.com/github/THEABHASHARMA/STATS-CONCEPTS-APPLIED-ON-REAL-WORLD-CASE-STUDIES-/blob/main/Sachin_case_study.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!gdown 1TwgJSuiUW8j3_tXsy6B8YwAzAY0tX4Jk

Downloading...
From: https://drive.google.com/uc?id=1TwgJSuiUW8j3_tXsy6B8YwAzAY0tX4Jk
To: /content/Sachin_ODI.csv
100% 26.4k/26.4k [00:00<00:00, 52.8MB/s]


In [ ]:
import numpy as np
import pandas as pd

In [ ]:
df = pd.read_csv("Sachin_ODI.csv")

In [ ]:
df.head()

runs  NotOut mins  bf  fours  sixes      sr  Inns           Opp    Ground  \
0    13       0   30  15      3      0   86.66     1   New Zealand    Napier   
1    37       0   75  51      3      1   72.54     2  South Africa  Hamilton   
2    47       0   65  40      7      0  117.50     2     Australia   Dunedin   
3    48       0   37  30      9      1  160.00     2    Bangladesh   Sharjah   
4     4       0   13   9      1      0   44.44     2      Pakistan   Sharjah   

         Date        Winner    Won  century  
0  1995-02-16   New Zealand  False    False  
1  1995-02-18  South Africa  False    False  
2  1995-02-22         India   True    False  
3  1995-04-05         India   True    False  
4  1995-04-07      Pakistan  False    False

In [ ]:
df.shape

(360, 14)

In [ ]:
#Q1. Out of 360 matches, how many matches did India Won?

In [ ]:
df["Won"].value_counts()

Won
True     184
False    176
Name: count, dtype: int64

In [ ]:
#Q2. A match is randomly chosen, what is the probability that India have won that match?

In [ ]:
df_won= df.loc[df["Won"] == True]
prob_winning=len(df_won)/len(df)
prob_winning

0.5111111111111111

In [ ]:
#Q3. A match is chosen at a random, what is the probability that Sachin has scored a Century in that match?

In [ ]:
df_century= df.loc[df["century"] == True]
prob_century=len(df_century)/len(df)
prob_century

0.12777777777777777

In [ ]:
#Q4. A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India have won that match?

In [ ]:
df_win_and_century = df[(df["Won"] == True) & (df["century"] == True)]
len(df_win_and_century)

30

In [ ]:
ans = len(df_win_and_century)/len(df)
ans

0.08333333333333333

In [ ]:
#Contigency table

pd.crosstab(index = df["century"],
            columns = df["Won"])

Won      False  True 
century              
False      160    154
True        16     30

In [ ]:
#Contigency table

pd.crosstab(index = df["century"],
            columns = df["Won"],
            margins = True)

Won      False  True  All
century                  
False      160   154  314
True        16    30   46
All        176   184  360

In [ ]:
#Q4. A match is chosen at a random. What is the probability that Sachin has scored a century in that match and India have won that match?

30/360

0.08333333333333333

Conclusion of the Problem statement:
Let's have a look how is Sachin's batting can or cannot impact the winning chances of India
1. Out of the 360 matches that Sachin has played, India have won 184 matches and Loose 176 matches.
2. So, if we choose any match at a random from Sachin's ODI career, there is a 51% chance that India have won that match.
3. Now, If we choose a random match from Sachin's ODI career, there is 12.77% chance that Sachin has scored a century in that match.
4. We know if a random match is choosen, there is 12.77% chance that Sachin has scored a century but
there is only 8% chance India have won that match.
we can conclude that the chances of India, Winning a match is more when Sachin didn't score a century (what an amazing insight)
Finally,
We can conclude that, if we pick a random match where Sachin played, India's win percentage is 51%. There is 12.77% chance of Sachin scoring a century in that match, and there is only 8% chance that in that match Sachin scores a century as well as India have won that match